jetty-home/pom.xml

```diff
@@ -337,7 +337,7 @@
             </goals>
             <configuration>
               <includeGroupIds>javax.annotation,org.eclipse.jetty.orbit,org.ow2.asm</includeGroupIds>
-              <includeArtifactIds>javax.annotation-api,asm,asm-commons</includeArtifactIds>
+              <includeArtifactIds>javax.annotation-api,asm,asm-commons,asm-tree,asm-analysis</includeArtifactIds>
```

Comment:
Why do you need to add these 2 jars? They are not necessary to run when jetty is run from the classpath.

Reply:
They were listed as required JPMS module names from within the `asm-commons` and `asm` definitions.

From `asm-6.2.jar`:

``` java
open module org.objectweb.asm {
    exports org.objectweb.asm;
    exports org.objectweb.asm.signature;
}
```

From `asm-commons-6.2.jar`:

``` java
open module org.objectweb.asm.commons {
    requires transitive org.objectweb.asm;
    requires transitive org.objectweb.asm.tree;
    requires transitive org.objectweb.asm.tree.analysis;

    exports org.objectweb.asm.commons;
}
```

Reply:
Well that's pretty rubbish. The asm commons jar doesn't need the tree stuff at all - we never deploy the tree jar for the distro or plugin etc.

Reply:
@janbartel unfortunately there is no way around. ASM has a proper `module-info.java` and Jetty won't even start if we don't put `asm.tree` and `asm.tree.analysis` in the module-path, as the JPMS module graph won't be resolved by the JVM if those jars are missing.